app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java

```diff
@@ -1359,5 +1376,62 @@ private void checkStateAndReconcile() throws UpfProgrammableException {
                 flowRuleService.apply(ops.build());
             }
         }
+
+        private void checkMeterStateAndReconcile() {
+            log.debug("Running meters reconciliation task...");
+            assertUpfIsReady(); // Use assertUpfIsReady to generate exception and log it on the caller
+
+            for (var entry : upfProgrammables.entrySet()) {
+                var deviceId = entry.getKey();
+                var upfProg = entry.getValue();
+                if (deviceId.equals(leaderUpfDevice)) {
+                    continue;
+                }
+                if (!mastershipService.isLocalMaster(deviceId)) {
+                    continue;
+                }
+                Set<Meter> leaderMeters =
+                        meterService.getMeters(leaderUpfDevice).stream()
+                                .filter(getLeaderUpfProgrammable()::fromThisUpf)
+                                .filter(m -> m.state() == MeterState.PENDING_ADD || m.state() == MeterState.ADDED)
+                                .collect(Collectors.toSet());
+                Set<Meter> followerMeters =
+                        meterService.getMeters(deviceId).stream()
+                                .filter(upfProg::fromThisUpf)
+                                .filter(m -> m.state() == MeterState.PENDING_ADD || m.state() == MeterState.ADDED)
+                                .map(m -> copyMeterForDevice(m, leaderUpfDevice))
+                                .collect(Collectors.toSet());
+
+                // Remove unexpected: Meter is in the follower but not in the leader
+                // Update stale: Meter is both on follower and leader but bands are different
+                // Add missing: Rule is in the leader but not in the follower
+
+                Set<Meter> unexpectedMeters = followerMeters.stream()
+                        .filter(fr -> leaderMeters.stream().noneMatch(lr -> lr.equals(fr)))
+                        .collect(Collectors.toSet());
+                followerMeters.removeAll(unexpectedMeters);
+
+                Set<Meter> staleMeters = leaderMeters.stream()
+                        .filter(lr -> followerMeters.stream().noneMatch(fr -> fr.equals(lr)
+                                && fr.bands().containsAll(lr.bands())))
```

Comment:
are we doing an assumption on the bands size ? Maybe worth to drop a comment as reminder

Reply:
added a check on the number of bands, this should solve the issue